### PR TITLE
Add reader list update form functionality

### PR DESCRIPTION
**Pull request by jsnmoon**
#### Changes proposed in this Pull Request

* Implements data layer logic for updating reader lists.
* Removes a now-unused wpcom-undocumented function.
* Makes the update form at `/read/list/:user/:slug/edit` functional.

#### Testing instructions

* Open a list at `/read/list/:user/:slug/edit`.
* Try updating the name, slug, privacy settings, and description.

### CODE REVIEW

File: client/lib/wpcom-undocumented/lib/undocumented.js [hunk 1]
```diff
@@ -1318,21 +1318,6 @@ Undocumented.prototype.readLists = function ( fn ) {
 	return this.wpcom.req.get( '/read/lists', { apiVersion: '1.2' }, fn );
 };
 
-Undocumented.prototype.readListsUpdate = function ( query, fn ) {
```
Comment (bluefuton) [inline client/lib/wpcom-undocumented/lib/undocumented.js:1321]: 🎉

**Comment by bluefuton** [comment]
This works well for updating title, description and public flag 👍 

I had problems updating the slug, because it seems to need the old slug to construct the URL for the API request. I think it's best to land this though, and address that in a followup.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3844619

Hi @jsnmoon, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `List updated successfully!`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3844619).

**Pull request merged**